components/org.wso2.carbon.identity.auth.attribute.handler/src/main/java/org/wso2/carbon/identity/auth/attribute/handler/AuthAttributeHandlerManager.java

```diff
@@ -89,6 +92,39 @@ public List<AuthAttributeHolder> getAvailableAuthAttributeHolders(String appId,
         return selectedAuthAttributeHolders;
     }
 
+    /**
+     * This method is used to engage the validations related to the
+     * selected auth attribute handler.
+     *
+     * @param authAttributeHandlerName Name of the auth attribute handler.
+     * @param attributeMap Set of attributes to be validated.
+     * @return A ValidationResult object containing the validation status.
+     * If there are validation failures it will contain a validation
+     * failure reason for any of the validation failed attributes.
+     * @throws AuthAttributeHandlerException authAttributeHandlerException.
+     */
+    public ValidationResult validateAuthAttributes(String authAttributeHandlerName, Map<String, String> attributeMap)
+            throws AuthAttributeHandlerException {
+
+        List<AuthAttributeHandler> authAttributeHandlers =
+                AuthAttributeHandlerServiceDataHolder.getInstance().getAuthAttributeHandlers();
+
+        AuthAttributeHandler authAttributeHandler = null;
+        for (AuthAttributeHandler handler : authAttributeHandlers) {
```

Comment:
can `authAttributeHandlers` be null in any case?

Reply:
Nope the list is initialized as an empty list in the very beginning so it cannot be null https://github.com/wso2-extensions/identity-governance/blob/master/components/org.wso2.carbon.identity.auth.attribute.handler/src/main/java/org/wso2/carbon/identity/auth/attribute/handler/internal/AuthAttributeHandlerServiceDataHolder.java#L34